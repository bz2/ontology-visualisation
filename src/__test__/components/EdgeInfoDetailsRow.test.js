import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EdgeInfoDetailsRow from '../../components/EdgeInfoDetailsRow'

const setup = () => {
  const props = {
    availableNodesNormalised: {
      abc: {
        id: 'abc',
        label: 'label',
        rdfAbout: 'rdfAbout',
        skosComment: 'skosComment',
        skosDefinition: 'skosDefinition',
        skosExample: 'skosExample'
      },
      cde: {
        id: 'cde',
        label: 'label2',
        rdfAbout: 'rdfAbout',
        skosComment: 'skosComment',
        skosDefinition: 'skosDefinition',
        skosExample: 'skosExample'
      }
    },
    connection: {
      from: 'abc',
      fromLabel: 'label',
      to: 'cde',
      toLabel: 'label2',
    },
  }

  const component = shallow(<EdgeInfoDetailsRow {...props} />)

  return {
    component,
    props
  }
}

describe('EdgeInfoDetailsRow', () => {
  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
