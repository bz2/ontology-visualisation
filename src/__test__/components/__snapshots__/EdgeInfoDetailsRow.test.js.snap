// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EdgeInfoDetailsRow should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "removeFromArray": [Function],
      "setStoreState": [Function],
    }
  }
  availableNodesNormalised={
    Object {
      "abc": Object {
        "id": "abc",
        "label": "label",
        "rdfAbout": "rdfAbout",
        "skosComment": "skosComment",
        "skosDefinition": "skosDefinition",
        "skosExample": "skosExample",
      },
      "cde": Object {
        "id": "cde",
        "label": "label2",
        "rdfAbout": "rdfAbout",
        "skosComment": "skosComment",
        "skosDefinition": "skosDefinition",
        "skosExample": "skosExample",
      },
    }
  }
  connection={
    Object {
      "from": "abc",
      "fromLabel": "label",
      "to": "cde",
      "toLabel": "label2",
    }
  }
  mapToProps={[Function]}
>
  <Component />
</Connect>
`;
